pkgs/top-level/python-packages.nix

```diff
@@ -2477,7 +2477,17 @@ in modules // {
     };
   };
 
-  boto3 = buildPythonPackage rec {
+  boto3 = let
+    # Version >=1.3,<1.4 is required by boto3.
+    botocore-1_3 = self.botocore.override rec {
+      name = "botocore-${version}";
+      version = "1.3.30";
+      src = pkgs.fetchurl {
+        url = "https://pypi.python.org/packages/6c/79/5d3abfb9fffe4a8cb0a113f088c1012743cf40db805aaac625b2e3cf2111/botocore-1.3.30.tar.gz";
```

Comment:
Fewer places for things to get out of sync. Unlike this one, a lot of URLs will work fine if you just update a version number, which means bumping a version involves bumping two strings (version and sha256) instead of three (version, sha256, and url). In terms of data models, I like to tell the computer when data is entangled, too.